src/algorithms/postburn/PostBurn.cc

```diff
@@ -0,0 +1,155 @@
+#include "algorithms/postburn/PostBurnConfig.h"
+#include "algorithms/reco/Beam.h"
+
+void eicrecon::PostBurn::init() {
+
+}
+
+void eicrecon::PostBurn::process(
+    const PostBurn::Input& input,
+    const PostBurn::Output& output) const {
+
+    const auto [mcparts] = input;
+    auto [outputParticles] = output;
+
+    bool      pidAssumePionMass = m_cfg.pidAssumePionMass;
+    double    crossingAngle    = m_cfg.crossingAngle;
+    double    pidPurity        = m_cfg.pidPurity;
+    bool      correctBeamFX    = m_cfg.correctBeamFX;
+    bool      pidUseMCTruth    = m_cfg.pidUseMCTruth;
+
+    bool      hasBeamHadron    = true;
+    bool      hasBeamLepton    = true;
+
+    //read MCParticles information for status == 1 particles and post-burn
+
+    ROOT::Math::PxPyPzEVector  e_beam(0.,0.,0.,0.);
+    ROOT::Math::PxPyPzEVector  h_beam(0.,0.,0.,0.);
+
+        auto incoming_lepton = find_first_beam_electron(mcparts);
+    if (incoming_lepton.size() == 0) {
+      debug("No beam electron found -- particleGun input");
+          hasBeamLepton = false;
+    }
+
+        auto incoming_hadron = find_first_beam_hadron(mcparts);
+    if (incoming_hadron.size() == 0) {
+      debug("No beam hadron found -- particleGun input");
+      hasBeamHadron = false;
+    }
+
+    if((hasBeamHadron && !hasBeamLepton) || (!hasBeamHadron && hasBeamLepton)){
+        debug("Only one beam defined! Not a possible configuration!");
+        return;
+    }
+
+        //handling for FF particle gun input!!
+    if(!hasBeamHadron || !hasBeamLepton){
+        for (const auto& p: *mcparts) {
+            if((p.getPDG() == 2212 || p.getPDG() == 2112)) { //look for "gun" proton/neutron
+                h_beam.SetPxPyPzE(crossingAngle*p.getEnergy(), 0.0, p.getEnergy(), p.getEnergy());
+                if(p.getEnergy() > 270.0 && p.getEnergy() < 280.0){
+                    e_beam.SetPxPyPzE(0.0, 0.0, -18.0, 18.0);
+                }
+            }
+        }
+    }
+        else{
+
+        if(correctBeamFX){
+
+            h_beam.SetPxPyPzE(incoming_hadron[0].getMomentum().x, incoming_hadron[0].getMomentum().y, incoming_hadron[0].getMomentum().z, incoming_hadron[0].getEnergy());
+            e_beam.SetPxPyPzE(incoming_lepton[0].getMomentum().x, incoming_lepton[0].getMomentum().y, incoming_lepton[0].getMomentum().z, incoming_lepton[0].getEnergy());
+
+        }
+        else{
+
+            h_beam.SetPxPyPzE(crossingAngle*incoming_hadron[0].getEnergy(), 0.0, incoming_hadron[0].getEnergy(), incoming_hadron[0].getEnergy());
+            e_beam.SetPxPyPzE(0.0, 0.0, -incoming_lepton[0].getEnergy(), incoming_lepton[0].getEnergy());
+
+        }
+    }
+
+
+
+    //Calculate boost vectors and rotations here
+
+    ROOT::Math::PxPyPzEVector cm_frame_boost = e_beam + h_beam;
+    ROOT::Math::PxPyPzEVector tmp(-cm_frame_boost.Px(), -cm_frame_boost.Py(), -cm_frame_boost.Pz(), cm_frame_boost.E());
+
+    ROOT::Math::Boost boostVector(tmp.Px()/tmp.E(), tmp.Py()/tmp.E(), tmp.Pz()/tmp.E());
```

Comment:
Can we get some nicer naming here, like so:
```suggestion
    ROOT::Math::Cartesian3D beta(-cm_frame_boost.Px() / cm_frame_boost.E(), -cm_frame_boost.Py() / cm_frame_boost.E(), -cm_frame_boost.Pz() / cm_frame_boost.E());

    ROOT::Math::Boost boostVector(beta);
```